### DISCUSSION TITLE
OpenSign on Synology NAS

### DISCUSSION THREAD

**Discussion by Dvalin21** [OpenSignLabs/OpenSign#1053] [Q&A]
Simply, what would be the easiest way to run this on a Synology NAS? I've tried wget the folder to the nas, run the commands to build (by the way, you need to update from docker compose to docker-compose for the updated dockers), and then it runs. Well at least I thought it was going to run. What happens is that the main url shows the favicon, but the page is blank. I don't see any errors from the logs.

**Comment by MrGuato** [comment]
Hey Dvalin21,

The blank page issue is often related to port configuration, so let's walk through the setup:

## Port Mapping in Docker:

Synology's Docker implementation requires explicit port mapping in your docker-compose.yml

Add this to your service configuration:
```
ports:
  - "8080:80"  # Maps container port 80 to NAS port 8080
```
You can change 8080 to any available port on your NAS

## Potential Port Conflicts:

If you want to use port 80, you'll need to:

Stop Synology Web Station (common conflict)

Check for other services using port 80 in Control Panel > Security > Firewall
Alternatively, stick with port 8080 or another unused port

## Firewall Configuration:

Go to Control Panel > Security > Firewall
If you don't immediately see Firewall, you might need to switch to "Advanced Mode" in the top-right menu
Create a new rule allowing inbound traffic to your chosen port (e.g., 8080)
Make sure the rule is enabled

## Troubleshooting:

Check Docker container logs for errors
Verify all containers are running with docker ps

Hope this helps!
